osu.Game.Tests/Visual/Online/TestSceneDirectPanel.cs

```diff
@@ -125,13 +130,38 @@ private void load(RulesetStore rulesets)
                         new GridBeatmapPanel(undownloadable),
                         new GridBeatmapPanel(manyDifficulties),
                         new GridBeatmapPanel(explicitMap),
+                        playDisabledPanel = new PlayDisabledGridBeatmapPanel(explicitMap),
                         new ListBeatmapPanel(normal),
                         new ListBeatmapPanel(undownloadable),
                         new ListBeatmapPanel(manyDifficulties),
                         new ListBeatmapPanel(explicitMap)
                     },
                 },
             };
+
+            AddStep("click disabled play button", () =>
+            {
+                InputManager.MoveMouseTo(playDisabledPanel.PlayButton);
+                InputManager.Click(MouseButton.Left);
+            });
+            AddAssert("nothing played", () => playDisabledPanel.PreviewPlaying.Value == false);
+        }
+
+        private class PlayDisabledGridBeatmapPanel : GridBeatmapPanel
+        {
+            public new PlayButton PlayButton => base.PlayButton;
+
+            public PlayDisabledGridBeatmapPanel(BeatmapSetInfo beatmap)
+                : base(beatmap)
+            {
+            }
+
+            protected override void LoadComplete()
+            {
+                base.LoadComplete();
+
+                PlayButton.ChildrenOfType<ClickableContainer>().Single().Enabled.Value = false;
```

Comment:
this also needs to go away. it should be no more difficult than directly setting `Enabled` on the button to test this.